### DISCUSSION TITLE
Database connection in Global Setup does not work

### DISCUSSION THREAD

**Discussion by EshaanAgg** [vitest-dev/vitest#2888] [Q&A] [answered]
I have configured `globalSetup.js` as a global setup script in my `vite.config.ts` and am using the same to connect to a Mongo database.

Namely, the code of `globalSetup.js` is :
```js
import "dotenv/config";
import mongoose from "mongoose";

export async function setup() {
  // Connect to the database and store the connection in a global variable
  global.MONGO_INSTANCE = await mongoose.connect(process.env.MONGO_DB_URL, {
    useCreateIndex: true,
    useUnifiedTopology: true,
    useFindAndModify: false,
    useNewUrlParser: true,
  });
}
```

After this, whenever I am trying to use any database operation in the tests, I am unable to use this Mongo connection. This is because the connection is being dropped. 

I have verified the same with the following snippet in `A.spec.ts` (which is a test file):
```ts
import { beforeAll } from "vitest";
import mongoose from "mongoose";

beforeAll(async () => {
  console.log(mongoose.connection.readyState);
});

```
Running this test file is logging `0` to the console, indicating that the database connection status is `disconnected`. Using the same console log at the end of the `setup` function in `globalSetup.js` prints `1`, which is used to show `connected` status. 

How can I debug this behaviour, as I want to run multiple test files with the same database connection.

**Comment by sheremet-va** [comment, marked as answer]
globalSetup runs in a different context and doesn't have access to the same global scope as your tests. You need to start mongo db there and connect to it inside each test file.

**Reply by EshaanAgg** [reply]
Can you please elaborate on how I might do the same?

One method is to call the `connect` method in each test file, which is tedious and in violation of the DRY principle.
Can I use the `setupFiles` configuration method to connect to the database using `mongoose.connect()`, or would I still face the same context error?

**Reply by sheremet-va** [reply]
Yes, you can. `setupFiles` runs in the same global scope as your tests.
